microsvcs/gpt_generation/README.md

```diff
@@ -0,0 +1,129 @@
+# api quick reference sheet
+## POST /api/scenario
+creates a new scenario for the user. 
+
+@params req.body.uid - required, user id <br>
+@params req.body.topic - optional, a selected or predefined topic to be used <br>
+
+Sample request body:
+```
+{
+    "uid":"666350518e5c4522aed85892", //required
+    "topic":"Diversity" //optional 
+}
+```
+
+Sample success response: 
+```
+{
+    "message": "Scenario Generated",
+    "success": true,
+    "content": {
+        "scenario": "I was facilitating a diversity and inclusion training session on the topic of gender inclusivity. During the session, there was a tense moment when a participant made a derogatory remark about non-binary individuals, which made some other participants uncomfortable.",
+        "options": {
+            "passive": "You should have ignored the remark and continued with the training, hoping that the discomfort would fade away.",
+            "reactive": "You should have addressed the remark immediately, pointing out its inappropriateness and its impact on creating a safe and inclusive environment for all participants.",
+            "proactive": "You should have set clear ground rules at the beginning of the training, emphasizing the importance of respect and inclusivity, to prevent such remarks from being made in the first place.",
+            "active": "You should have used the moment as a teaching opportunity, engaging the group in a discussion about gender diversity, asking the participant to elaborate on their remark and guiding them towards understanding and empathy."
+        }
+    }
+}
+
+```
+
+### Additional Info:
+This API's current behaviour has a behavioural lapse for the microservice. Only this endpoint can create a new Thread
+for the user to use, meaning that the other endpoints will fail if the POST /api/scenario is not called AND if the user does not have a record in the DB.
+
+Sample DB Record:
+```
+{
+  "_id": {
+    "$oid": "66673f1e177bd5f9a52f3ca7"
+  },
+  "uid": "666350518e5c4522aed85892",
+  "thread_info": {
+    "id": "thread_0BRNOfWhaDZrBAAI3xNv4OxB",
+    "object": "thread",
+    "created_at": 1718042398,
+    "metadata": {
+      "uid": "666350518e5c4522aed85892"
+    }
+  },
+  "createdAt": {
+    "$date": "2024-06-10T17:59:59.000Z"
+  },
+  "updatedAt": {
+    "$date": "2024-06-10T17:59:59.000Z"
+  },
+  "__v": 0
+}
+```
+
+---
+## PUT /api/scenario/end
```

Comment:
PUT /api/scenario is fine, unless you really intend this to be the final response for the scenario